### PR TITLE
Support service registration retry

### PR DESCRIPTION
**Pull request by koalalam**

### CODE REVIEW

**Comment by jplock** [comment]
Thanks for this PR.  If present, can we shutdown the scheduled executor service in https://github.com/smoketurner/dropwizard-consul/blob/master/consul-core/src/main/java/com/smoketurner/dropwizard/consul/managed/ConsulAdvertiserManager.java somehow as well?

**Review by jplock — CHANGES_REQUESTED** [review]
Thanks for this PR. If present, can we shutdown the scheduled executor service in https://github.com/smoketurner/dropwizard-consul/blob/master/consul-core/src/main/java/com/smoketurner/dropwizard/consul/managed/ConsulAdvertiserManager.java somehow as well?

File: consul-core/src/main/java/com/smoketurner/dropwizard/consul/core/ConsulServiceListener.java [hunk 1]
```diff
@@ -15,27 +15,64 @@
...
   public void serverStarted(final Server server) {
     final int applicationPort = getLocalPort(server);
     final int adminPort = getAdminPort(server);
-    advertiser.register(applicationPort, adminPort);
+    register(applicationPort, adminPort);
+  }
+
+  void register(int applicationPort, int adminPort) {
+    try {
+      advertiser.register(applicationPort, adminPort);
```
Comment (koalalam) [inline consul-core/src/main/java/com/smoketurner/dropwizard/consul/core/ConsulServiceListener.java:61]: @jplock , do you mean shutting down scheduler (if present) here?  when there is no exception meaning the service registration is completed successfully?

Comment (jplock) [reply consul-core/src/main/java/com/smoketurner/dropwizard/consul/core/ConsulServiceListener.java:61]: no, I meant ensuring the scheduler is shutdown when shutting down the application

Comment (koalalam) [reply consul-core/src/main/java/com/smoketurner/dropwizard/consul/core/ConsulServiceListener.java:61]: Got it. Make sense. Will add it.

**Review by jplock — APPROVED** [review]

**Pull request merged**